src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceUpdateSettings.cs

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Configuration
+{
+    public class PackageSourceUpdateSettings
+    {
+        public bool UpdateCredentials { get; set; }
+        public bool UpdateEnabled { get; set; }
+
+        public PackageSourceUpdateSettings(bool updateCredentials = true, bool updateEnabled = true)
```

Comment:
In general I'm not a fan of optional arguments. If in the future we add a 3rd property to this class, adding a 3rd parameter to this constructor will be a breaking change (binary compatibility, not compile-time compatibility), but adding an overload will violate [PublicApiAnalyzer rule RS0026 and/or RS0027](https://github.com/dotnet/roslyn-analyzers/blob/master/src/PublicApiAnalyzers/Microsoft.CodeAnalysis.PublicApiAnalyzers.md).